### PR TITLE
feat: manage require_last_push_approval and allow_deletions in branch protection

### LINKED ISSUE #2360: Feature request. Support require_last_push_approval
require_last_push_approval 
is part of branch protection under pull requests. Its currently not supported by PyGithub.

### PR DESCRIPTION
**Pull request by terenho**
fixes #2360

### CODE REVIEW

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Can we have tests for the new fields?

File: github/Branch.py [hunk 1]
```diff
@@ -134,6 +134,7 @@ def edit_protection(
         teams_bypass_pull_request_allowances: Opt[list[str]] = NotSet,
         apps_bypass_pull_request_allowances: Opt[list[str]] = NotSet,
         block_creations: Opt[bool] = NotSet,
+        require_last_push_approval: Opt[bool] = NotSet,
     ) -> None:
```
[the remark this answers is not in the document]
Comment (terenho) [reply github/Branch.py:139]: Added `allow_deletions` in the latest commits

File: github/BranchProtection.py [hunk 2]
```diff
@@ -55,6 +55,7 @@ def _initAttributes(self) -> None:
         self._required_pull_request_reviews: Attribute[RequiredPullRequestReviews] = NotSet
         self._user_push_restrictions: Opt[str] = NotSet
         self._team_push_restrictions: Opt[str] = NotSet
+        self._require_last_push_approval: Attribute[bool] = NotSet
```
[the remark this answers is not in the document]
Comment (terenho) [reply github/BranchProtection.py:59]: Added `allow_deletions` in the latest commits

**Comment by terenho** [comment]
> Can we have tests for the new fields?

Added/updated tests for the new fields in the latest commits

**Comment by terenho** [comment]
@EnricoMi I have added the allow_deletion function and added tests, I also resolved a merge conflict. This PR is ready for review.

**Comment by EnricoMi** [comment]
Superseded by #2873, added you as co-author to appreciate your contribution.

**Pull request closed without merging**